### ISSUE TITLE
Maps of 1024x1024 (and higher) result in error / Redo drawing of map in Viewport

### ISSUE THREAD

**Issue by stefanhendriks** [stefanhendriks/dune2themaker4j#15]
The drawing logic is now allocating a texture of `screenWidth`x`screenHeight`. Which results in an error:
```
Sun Jan 25 21:17:01 CET 2015 ERROR:Attempt to allocate a texture to big for the current hardware
java.io.IOException: Attempt to allocate a texture to big for the current hardware
	at org.newdawn.slick.opengl.InternalTextureLoader.getTexture(InternalTextureLoader.java:431)
	at org.newdawn.slick.opengl.InternalTextureLoader.createTexture(InternalTextureLoader.java:374)
	at org.newdawn.slick.Image.<init>(Image.java:237)
	at org.newdawn.slick.Image.<init>(Image.java:221)
	at com.fundynamic.d2tm.game.map.MapRenderer.makeImage(MapRenderer.java:25)
```

To solve this, the drawing should be done by the respective Viewport and it should only draw that part it sees (instread of drawing *all* and then simply taking a piece of the 'terrain texture').

Especially since we want to support MUCH BIGGER maps than this.

**Comment by stefanhendriks** [comment]
Shroud rendering is done in the MapRenderer - which feels quirky, probably because we do some stuff weird now. This issue should be top of list before doing anything else. Shroud rendering has a bit different facing logic but looks a lot the same. Perhaps this can be done more efficiently. See https://github.com/stefanhendriks/dune2themaker4j/pull/16 for more details.

**Comment by stefanhendriks** [comment]
Drawing has been redone, very crude implementation. But, it can now view maps up to 4096x4096 (probably even bigger, but did not try it).

Several thoughts though:
- [ ] what is that with cells? Are they creating images? Or are they just taking things from the `Theme`? Gotta go back and figure out what is happening there.
- [ ] drawing can be done with an own abstraction (`TerrainDrawer`, `ShroudDrawer`, etc?)

**Comment by stefanhendriks** [comment]
Fixed by #19
